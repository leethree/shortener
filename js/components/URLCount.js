/* @flow */

import React from 'react';
import { graphql, createFragmentContainer } from 'react-relay';
import { Text, Box } from 'rebass';

const URLCount = ({ viewer }) => {
  const numUrls = viewer.totalCount;
  return (
    <Box color="#888888">
      {numUrls === 0 ? (
        <Text textAlign="right">Shorten URL above</Text>
      ) : (
        <Text textAlign="right">
          <strong>{numUrls}</strong> URL{numUrls === 1 ? '' : 's'} shortened
        </Text>
      )}
    </Box>
  );
};

export default createFragmentContainer(
  URLCount,
  graphql`
    fragment URLCount_viewer on User {
      id
      totalCount
    }
  `,
);
