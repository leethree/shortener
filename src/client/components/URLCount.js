/* @flow */

import React from 'react';
import { graphql, createFragmentContainer } from 'react-relay';
import { Text, Box } from 'rebass';

import { type URLCount_viewer } from './__generated__/URLCount_viewer.graphql';

type Props = {
  viewer: URLCount_viewer,
};

const URLCount = ({ viewer }: Props) => {
  const numUrls = viewer.totalCount;
  return (
    <Box color="#888888">
      {numUrls && numUrls <= 10 ? (
        <Text textAlign="right">
          <strong>{numUrls}</strong> URL{numUrls === 1 ? '' : 's'} shortened
        </Text>
      ) : (
        <Text textAlign="right">
          Showing 10 of <strong>{numUrls}</strong> URLs shortened
        </Text>
      )}
    </Box>
  );
};

export default createFragmentContainer(
  URLCount,
  graphql`
    fragment URLCount_viewer on User {
      id
      totalCount
    }
  `,
);
