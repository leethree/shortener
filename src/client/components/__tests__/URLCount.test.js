import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import URLCount from '../URLCount';

test.each([0, 1, 2, 10, 11])('URLCount renders count=%d', count => {
  const viewer = {
    id: 'me',
    totalCount: count,
  };
  const component = renderer.create(<URLCount viewer={viewer} />);
  expect(component).toMatchSnapshot();
});
