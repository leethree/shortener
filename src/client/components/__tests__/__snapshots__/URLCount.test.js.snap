// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`URLCount renders count=0 1`] = `
.c0 {
  box-sizing: border-box;
  color: darken;
}

.c1 {
  margin: 0px;
  text-align: right;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <strong>
      0
    </strong>
     
    URLs
     shortened
  </div>
</div>
`;

exports[`URLCount renders count=1 1`] = `
.c0 {
  box-sizing: border-box;
  color: darken;
}

.c1 {
  margin: 0px;
  text-align: right;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <strong>
      1
    </strong>
     
    URL
     shortened
  </div>
</div>
`;

exports[`URLCount renders count=2 1`] = `
.c0 {
  box-sizing: border-box;
  color: darken;
}

.c1 {
  margin: 0px;
  text-align: right;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <strong>
      2
    </strong>
     
    URLs
     shortened
  </div>
</div>
`;

exports[`URLCount renders count=10 1`] = `
.c0 {
  box-sizing: border-box;
  color: darken;
}

.c1 {
  margin: 0px;
  text-align: right;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <strong>
      10
    </strong>
     
    URLs
     shortened
  </div>
</div>
`;

exports[`URLCount renders count=11 1`] = `
.c0 {
  box-sizing: border-box;
  color: darken;
}

.c1 {
  margin: 0px;
  text-align: right;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    Showing 10 of 
    <strong>
      11
    </strong>
     URLs shortened
  </div>
</div>
`;
